File: workspace/src/code/load-greeting.ts
import type { GreetingApi } from "../contracts/greeting-api";

export async function loadGreeting(name: string, api: GreetingApi) {
  const response = await api.getGreeting(name);

  return {
    submittedName: name,
    message: response.message,
  };
}

File: workspace/lib/code/load_greeting.dart
import '../contracts/greeting_api.dart';

class GreetingViewModel {
  final String submittedName;
  final String message;

  const GreetingViewModel({required this.submittedName, required this.message});
}

Future<GreetingViewModel> loadGreeting(String name, GreetingApi api) async {
  final submittedName = name.trim();
  final response = await api.getGreeting(submittedName);

  return GreetingViewModel(
    submittedName: submittedName,
    message: response.message,
  );
}
